### ISSUE TITLE
Extension causes high cpu load

### ISSUE THREAD

**Issue by tanmoy02** [angular/vscode-ng-language-service#2037]
- Issue Type: `Performance`
- Extension Name: `ng-template`
- Extension Version: `18.0.0`
- OS Version: `Windows_NT x64 10.0.22631`
- VS Code version: `1.89.1`

![image](https://github.com/angular/vscode-ng-language-service/assets/1296341/e9de9080-0f44-4991-bd71-679ed40aa689)

[Angular.ng-template-unresponsive.cpuprofile.txt](https://github.com/angular/vscode-ng-language-service/files/15481800/Angular.ng-template-unresponsive.cpuprofile.txt)

**Comment by atscott** [comment]
@tanmoy02 Can you provide more information? 

* Is there a particular workflow you're performing that causes the issue?
* Are you able to pinpoint this issue to a particular version of the extension? 
* Is the issue persistent or does it go away?

**Comment by tanmoy02** [comment]
@atscott No, there's no specific workflow to mention. I was working on a service file in my Angular 16 app, and the editor started lagging. After checking the status of running extensions, I noticed that this one and another code completion extension were unresponsive. I disabled the code completion extension first and restarted my editor, but the lag persisted, and this extension was still unresponsive. After disabling all unresponsive extensions, the performance improved slightly.
I'm using VS Code (1.89.1).

**Comment by atscott** [comment]
@tanmoy02 Can you try installing some older versions of the extension? My suspicion is that this happened in 17.3.2 and you won't see the problem in 17.3.1

**Comment by tanmoy02** [comment]
> @tanmoy02 Can you try installing some older versions of the extension? My suspicion is that this happened in 17.3.2 and you won't see the problem in 17.3.1

Okay, but 17.3.1 is not available, should I install 17.3.0?
https://github.com/angular/vscode-ng-language-service/releases

**Comment by atscott** [comment]
17.3.1 was published to the marketplace (https://marketplace.visualstudio.com/items?itemName=Angular.ng-template). I just forgot to put it in the releases in the repo

**Comment by tanmoy02** [comment]
@ivanwonder Thank you for your guidance.
@atscott I have installed 17.3.1 from the marketplace & so far it is working without any issues.

**Comment by atscott** [comment]
@tanmoy02 Okay, thank you for taking a look. @ivanwonder I think sadly we may need to revert https://github.com/angular/vscode-ng-language-service/commit/13d9776143dc6e7941005763a09211e01ed692a5
